src/components/date-input/DateInput.tsx

```diff
@@ -8,34 +8,41 @@ import React, {
   SyntheticEvent,
 } from 'react';
 import classNames from 'classnames';
+import ErrorMessage from '../error-message';
+import Hint from '../hint';
+import Label from '../label';
 
 interface IDateInputContext {
   isDateInput: boolean;
   registerRef: (type: DateInputType, ref: HTMLInputElement | null) => void;
   name: string;
   autoCompletePrefix: string | undefined;
+  error?: string
 }
 
 const DateInputContext = createContext<IDateInputContext>({
   isDateInput: false,
-  registerRef: () => {},
+  registerRef: () => undefined,
   name: '',
   autoCompletePrefix: '',
+  error: '',
 });
 
 type DateInputType = 'Day' | 'Month' | 'Year';
 interface DateInputInputProps extends HTMLProps<HTMLInputElement> {
   dateInputType: DateInputType;
+  errorFromField?: boolean
```

Comment:
We could rename this prop just to `error`, then don't pull `error` out of the props when they're destructured